### PR TITLE
Some cleanup

### LINKED ISSUE #509: git describe in repo with no tags produces fatal error message
https://github.com/terrapower/armi/blob/72dd0724278a31fe7b843a1e56db1c7f8ac04f60/armi/bookkeeping/db/database3.py#L638
can produce an ominous message if running in a repository that doesn't have any tags
```
fatal: No names found, cannot describe anything.
```

This can be demonstrated with
```shell
$ git init simple
$ cd simple
$ git describe
fatal: No names found, cannot describe anything.
```
Fix appears to be piping stderr to `/dev/null` like
https://github.com/terrapower/armi/blob/72dd0724278a31fe7b843a1e56db1c7f8ac04f60/armi/bookkeeping/db/database3.py#L631-L632

### PR DESCRIPTION
**Pull request by john-science**
## Description

This PR does several things:

* general cleanup
* removing unused "blog" section from the docs
* closes [this ticket](https://github.com/terrapower/terrapower.github.io/issues/6) where some "Edit on GitHub" links were broke in the docs
* closes #509 , where `grabLocalCommitHash()` could fail in some forked repos

---

## Checklist

- [x] Tests have been added/updated to verify that the new or changed code works.
- [X] Docstrings were included and updated as necessary
- [X] The code is understandable and maintainable to people beyond the author
- [X] There is no commented out code in this PR.

If user exposed functionality was added/changed:

- [X] Documentation added/updated in the `doc` folder.
- [X] New or updated dependencies have been added to `setup.py`.

### CODE REVIEW

**Comment by john-science** [comment]
@keckler  Note: this PR solves https://github.com/terrapower/terrapower.github.io/issues/6
@drewejohnson Note: This PR solves https://github.com/terrapower/armi/issues/509

**Comment by john-science** [comment]
@ntouran  Since this PR is not just one _thing_, please don't "Squash and Merge" this PR.  You can leave the Merge to me though. Thanks!

**Comment by john-science** [comment]
@ntouran  If you nuke your `/.tox/` directory, and do `tox -e doc`, this PR gets us pretty close to having a clean Sphinx build with no warnings. There are still some `FutureWarning`s, straight from Sphinx. But still. We're getting closer.

**Review by ntouran — APPROVED** [review]
only thing I don't really get is the explicit numbering.

File: doc/release/0.1.rst [hunk 1]
```diff
@@ -9,80 +9,75 @@ Release Date: 2021-08-09
...
-  to eliminate a limitation on how many i-indices and k-indices were allowed. This will
-  invalidate any user setting value that includes a location label (e.g. in
-  ``detailAssemsByBOLLocation``). A migration script may be used to assist in migration.
-
-* Removed the ``localization`` module, and shifted most of that exception handling to less
-  custom exception types. Though, there were also some functions moved to:
-  ``armi.utils.customExceptions.py``.
-
-* ``Settings`` are now immutable (or nearly so).
+1. ``geomType`` arguments in most places has been changed to accept a ``GeomType``
```
[the remark this answers is not in the document]
Comment (john-science) [reply doc/release/0.1.rst:12]: Oh! That's a good tip! I'll do that. 

Thanks, Sphinx Wizard.

**Pull request merged**